src/test/java/seedu/address/model/good/GoodQuantityTest.java

```diff
@@ -0,0 +1,46 @@
+package seedu.address.model.good;
+
+import static org.junit.jupiter.api.Assertions.assertFalse;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+import static seedu.address.logic.commands.CommandTestUtil.VALID_GOOD_QUANTITY_ONE;
+import static seedu.address.logic.commands.CommandTestUtil.VALID_GOOD_QUANTITY_ZERO;
+import static seedu.address.testutil.Assert.assertThrows;
+
+import org.junit.jupiter.api.Test;
+
+public class GoodQuantityTest {
+
+    @Test
+    public void constructor_invalidName_throwsIllegalArgumentException() {
+        int invalidGoodQuantity = -1;
+        assertThrows(IllegalArgumentException.class, () -> new GoodQuantity(invalidGoodQuantity));
+    }
+
+    @Test
+    public void isValidGoodQuantity() {
+        // invalid quantity
+        assertFalse(GoodQuantity.isValidGoodQuantity(-1)); // negative value
+
+        // valid quantity
+        assertTrue(GoodQuantity.isValidGoodQuantity(0));
+        assertTrue(GoodQuantity.isValidGoodQuantity(1));
+        assertTrue(GoodQuantity.isValidGoodQuantity(+1)); // positive value with plus sign
```

Comment:
What is the rationale for allowing this?

Reply:
I do not really understand your question, but the idea is ensure that the quantity must be >=0